modules/fitsfilereader/rendering/renderabletimevaryingfitssphere.cpp

```diff
@@ -0,0 +1,710 @@
+        {static_cast<int>(TextureFilter::Linear), "Linear smoothing"}
+        });
+    if (p.textureFilter.has_value()) {
+        _textureFilterProperty =
+            static_cast<int>(codegen::map<TextureFilter>(*p.textureFilter));
+    }
+    else {
+        _textureFilterProperty = static_cast<int>(TextureFilter::NearestNeighbor);
+    }
+    _renderForever = p.showPastFirstAndLastImage.value_or(_renderForever);
+
+    _textureFilterProperty.onChange([this]() {
+        switch (_textureFilterProperty) {
+        case(0):
+            for (auto file = _files.begin(); file != _files.end(); ++file) {
+                file->texture->setFilter(ghoul::opengl::Texture::FilterMode::Nearest);
+            }
+            break;
+        case(1):
+            for (auto file = _files.begin(); file != _files.end(); ++file) {
+                file->texture->setFilter(ghoul::opengl::Texture::FilterMode::Linear);
+            }
+            break;
+        }
+    });
+    addProperty(_textureFilterProperty);
+
+    _fitsLayer.onChange([this]() {
+        if (_loadingType == LoadingType::StaticLoading) {
+            extractMandatoryInfoFromSourceFolder();
+        }
+        else {
+            if (_isFitsFormat) {
+                for (auto file = _files.begin(); file != _files.end(); ++file) {
+                    std::pair<float, float> minMax = _layerMinMaxCaps.at(_fitsLayer);
+                    file->texture =
+                        loadTextureFromFits(file->path, _fitsLayer, minMax);
+                    file->texture->uploadTexture();
+                    file->texture->setFilter(ghoul::opengl::Texture::FilterMode::Nearest);
+                }
+                loadTexture();
+            }
+        }
+    });
+
+    _fitsLayerName.onChange([this]() {
+        if (_loadingType == LoadingType::StaticLoading) {
+            extractMandatoryInfoFromSourceFolder();
+        }
+        else {
+            if (_isFitsFormat) {
+                for (auto file = _files.begin(); file != _files.end(); ++file) {
+                    std::pair<float, float> minMax = _layerMinMaxCaps.at(_fitsLayerName);
+                    file->texture =
+                        loadTextureFromFits(file->path, _fitsLayerName, minMax);
+                    file->texture->uploadTexture();
+                    file->texture->setFilter(ghoul::opengl::Texture::FilterMode::Nearest);
+                }
+                loadTexture();
+            }
+        }
+    });
+
+    if (_loadingType == LoadingType::StaticLoading) {
+        extractMandatoryInfoFromSourceFolder();
+        computeSequenceEndTime();
+        loadTexture();
+    }
+    if (_loadingType == LoadingType::DynamicDownloading) {
+        _dataID = p.dataID.value_or(_dataID);
+        if (!_dataID) {
+            throw ghoul::RuntimeError(
+                "If running with dynamic downloading, dataID needs to be specified"
+            );
+        }
```

Comment:
```suggestion
        if (!p.dataID.has_value()) {
            throw ghoul::RuntimeError(
                "If running with dynamic downloading, dataID needs to be specified"
            );
        }
        
        _dataID = *p.dataID;
```

Same for all of the other values in here.  See suggestions from the previous times